### ISSUE TITLE
Auto-completion of property names fails when capturing output from Out-GridView command

### ISSUE THREAD

**Issue by pcgeek86** [PowerShell/vscode-powershell#5055]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [_open and closed_ issues](https://github.com/PowerShell/vscode-powershell/issues?q=is%3Aissue) to ensure it has not already been reported.
- [X] I have read the [troubleshooting](https://github.com/PowerShell/vscode-powershell/blob/main/docs/troubleshooting.md) guide.
- [X] I am sure this issue is with the _extension itself_ and does not reproduce in a standalone [PowerShell](https://github.com/PowerShell/PowerShell/issues/new/choose) instance.
- [X] I have verified that I am using the latest version of Visual Studio Code and the PowerShell extension.
- [ ] If this is a security issue, I have read the [security issue reporting guidance](https://github.com/PowerShell/vscode-powershell/blob/main/SECURITY.md).

### Summary

When a variable captures output, that is being piped through the `Out-GridView` command, the VSCode PowerShell extension is unable to auto-complete the property names from that variable.

The declared `[OutputType()]`  seems to get "lost" from the `Get-AzStorageAccount` command, when it's piped through `Out-GridView`. Is there a way to preserve this?

### PowerShell Version

```console
Name                           Value
----                           -----
PSVersion                      7.4.5
PSEdition                      Core
GitCommitId                    7.4.5
OS                             Microsoft Windows 10.0.22631
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

Name             : Visual Studio Code Host
Version          : 2024.2.2
InstanceId       : 652c5855-aad9-4300-a9bd-6f429676af8d
UI               : System.Management.Automation.Internal.Host.InternalHostUserInterface
CurrentCulture   : en-US
CurrentUICulture : en-US
PrivateData      : Microsoft.PowerShell.ConsoleHost+ConsoleColorProxy
DebuggerEnabled  : True
IsRunspacePushed : False
Runspace         : System.Management.Automation.Runspaces.LocalRunspace
```

### Visual Studio Code Version

```console
1.93.1
38c31bc77e0dd6ae88a4e9cc93428cc27a56ba40
x64
```

### Extension Version

```console
ms-vscode.powershell@2024.2.2
```

### Steps to Reproduce

1. Use code block below
2. Try to auto-complete the property name on line 2 (CTRL + SPACE after period)
3. It fails
4. Remove the pipe to `Out-GridView`
5. Try auto-completing the property name on line 2 again
6. It works
7. See video reproduction below

```pwsh
$SelectedStorageAccount = Get-AzStorageAccount | Out-GridView -Title 'Select an Azure Storage Account' -OutputMode Single
$SelectedStorageAccount.ResourceGroupName
```

### Visuals

https://github.com/user-attachments/assets/ef13a1ac-05d0-4486-951d-221848dd15c1

**Comment by JustinGrote** [comment]
Thanks for your submission!

It works fine for me, the vscode extension uses the same completion engine powershell does, so you have to run the first line before the runtime form of $test is known.
![image](https://github.com/user-attachments/assets/33a2aa0b-e5f9-4baa-9c32-34451c76bcd5)

If you want completion without having to run the command in the extension terminal first, you have to type the variable, because PowerShell has no idea what type Out-Gridview will emit. 
![image](https://github.com/user-attachments/assets/a0b898a7-7f18-4485-a9da-9d1f2962cca8)


IF you want that to be inferred without running the command, that has to be "fixed" in PowerShell itself (if even possible).

**Comment by pcgeek86** [comment]
Thanks @JustinGrote! I appreciate the screenshots and detailed description.

You're right that it works if the variable has been defined in the PowerShell session.

I did test this out in PowerShell, using Windows Terminal, but the obvious difference is that I had actually defined the variable there. Whereas, in the VSCode repro, I was not actually defining the variable. I didn't think about that.

Anyway, I can file this feedback in PowerShell instead of VSCode PowerShell extension! Much appreciated.
